model/project_aliases.go

```diff
@@ -236,6 +261,71 @@ func FindAliasInProjectRepoOrConfig(projectID, alias string) ([]ProjectAlias, er
 	return findMatchingAliasForProjectConfig(projectID, alias)
 }
 
+// FindMergedAliasesFromProjectRepoOrProjectConfig returns all aliases tht would be used by this project ref configuration,
+// by merging together repo and config aliases, if defined by the project.
+// Sets source equal to where the alias was defined.
+func FindMergedAliasesFromProjectRepoOrProjectConfig(projectRef *ProjectRef, projectConfig *ProjectConfig) ([]ProjectAlias, error) {
+	if projectRef == nil {
+		return nil, nil
+	}
+	projectAliases, err := FindAliasesForProjectFromDb(projectRef.Id)
+	if err != nil {
+		return nil, errors.Wrap(err, "finding project aliases")
+	}
+	aliasesToReturn := map[string]ProjectAliases{}
+	const patchAliasKey = "patch_alias"
+	for _, alias := range projectAliases {
+		aliasName := alias.Alias
+		if IsPatchAlias(aliasName) {
+			aliasName = patchAliasKey
+		}
+		alias.Source = AliasSourceProject
+		aliasesToReturn[aliasName] = append(aliasesToReturn[aliasName], alias)
+	}
+	// All aliases are covered in the project, so there's no reason to look at other sources
+	aliasesCovered := allAliasesCovered(aliasesToReturn)
```

Comment:
Can we instead return a list of alias names that are not covered instead of checking if we have all 5 alias names in the map? Then we can just check for these missing alias names in the loop below and we can remove the `if len(aliasesToReturn[aliasName]) == 0 || ...` checking